game/src/test/java/com/hiddenswitch/spellsource/CustomCardsTests.java

```diff
@@ -7409,4 +7409,18 @@ public void testWardenSaihan() {
 			assertEquals(warden2.getHp(), warden2.getMaxHp());
 		}));
 	}
+
+	@Test
+	public void testStringShot() {
+		runGym(((context, player, opponent) -> {
+			playCard(context, opponent, "minion_test_3_2");
+			receiveCard(context, player, "spell_string_shot");
+			player.setMana(1);
+			int originalNumberOfActions = context.getValidActions().size();
+
+			playCard(context, player, "minion_lil_wormy");
+			assertEquals(context.getValidActions().size(), originalNumberOfActions + 1);
```

Comment:
for these sorts of things, it's better to do
`assertEquals(context.getValidActions().stream().filter(action -> Objects.equals(action.getSourceReference(), stringShotCard) && Objects.equals(action.getTargetReference(), lilWormy)).count(), 1L)`

 - observe we can check if an action uses the appropriate card and target without doing too much additional inspection
 - observe the use of `Objects.equals` so we handle nulls correctly (`.equals` will throw on a null, and `EndTurnAction` has nulls)
 - observe the use of `1L` because counts are longs and the comparison will **fail** between an `int` and a `long` in my experience

Reply:
Makes sense. I'll try to do this sort of thing more going forward.